packages/stream/solana/StreamClient.ts

```diff
@@ -170,6 +196,96 @@ export class SolanaStreamClient extends BaseStreamClient {
     return { ixs, txId: signature, metadataId: metadataPubKey.toBase58() };
   }
 
+  async prepareCreateAlignedUnlockInstruction(
+    recipient: IRecipient,
+    streamParams: IStreamConfig,
+    extParams: ICreateStreamSolanaExt,
+  ): Promise<{ ixs: TransactionInstruction[]; metadata: Keypair | undefined; metadataPubKey: PublicKey }> {
+    const {
+      tokenId: mint,
+      start,
+      period,
+      cliff,
+      canTopup,
+      cancelableBySender,
+      cancelableByRecipient,
+      transferableBySender,
+      transferableByRecipient,
+      partner,
+    } = streamParams;
+    const { isNative, sender, computeLimit, computePrice, metadataPubKeys, alignedConfigParams } = extParams;
+
+    if (!sender.publicKey) {
+      throw new Error("Sender's PublicKey is not available, check passed wallet adapter!");
+    }
+
+    const { minPrice, maxPercentage, minPercentage, maxPrice, oracleType, skipInitial, tickSize, priceOracle } =
+      alignedConfigParams!;
```

Comment:
unsafe. why so?

Reply:
This method only gets called if `alignedConfigParams` are defined in the 'create' methods, but now that I think about it we expose this function in the client so it would make sense to add a check here as well

Reply:
> This method **only** gets called if
imo it is a wrong assumption. What if I call it from outside of the service itself?

new StreamClient().prepareCreateAlignedUnlockInstruction() from my codebase?

Reply:
> but now that I think about it we expose this function in the client so it would make sense to add a check here as well

I agreed :D